src/Sema.zig

```diff
@@ -11754,18 +11749,14 @@ pub fn resolveDeclFields(sema: *Sema, block: *Scope.Block, src: LazySrcLoc, ty:
     switch (ty.tag()) {
         .@"struct" => {
             const struct_obj = ty.castTag(.@"struct").?.data;
-            if (struct_obj.owner_decl.namespace != sema.owner_decl.namespace) return;
+            if (struct_obj.owner_decl.namespace.parent != sema.owner_decl.namespace) return;
```

Comment:
I don't understand what this is trying to check.  I had to change it to keep the tests passing, but I'm not sure if it's still correct.  @andrewrk can you please explain what this is for?

Reply:
This was trying to check that the sema was the correct one for resolving the struct fields.  Since we now use captures for values that cross this boundary, the whole struct resolution stack is no longer needed.  It's been deleted in this PR.